File: lib/con_math.c
#include "con_math.h"

unsigned int primeTestMax(const unsigned int number) {
    // returns the maximum value that should be tested when iterating of potential factors
    double root = sqrt((double)number);
    return (unsigned int)ceil(root);
}

bool isPrimeNaive(const unsigned int number) {
    // uses naive iterative method to test if a number is prime
    if (number == 1 || number % 2 == 0)
        return false;

    if (number == 2)
        return true;

    unsigned int testMax = primeTestMax(number), divisor;

    for(divisor = 3; divisor <= testMax; divisor += 2)
        if (number % divisor == 0)
            return false;

    return true;
}

File: pythonlib/con_math.py
from math import ceil

def prime_test_max(number):
    return int(ceil(number ** 0.5))

def is_prime_naive(number):
    if number == 1:
        return False

    if number == 2:
        return True

    if number % 2 == 0:
        return False

    test_max = prime_test_max(number)

    for divisor in xrange(3, test_max + 1, 2):
        if number % divisor == 0:
            return False

    return True
